### ISSUE TITLE
Solution status when using a mip gap or time limit

### ISSUE THREAD

**Issue by MikeSteglich** [ERGO-Code/HiGHS#1147]
When a MIP is solved with the command line tool with a mip gap, the solution status is expressed as `INTEGER OPTIMAL` in the solution file. Wouldn't it be better to output this status (as with SCIP, for example) as `MIP GAP REACHED` or `INTEGER FEASIBLE`?

If it is solved with time_limit and a solution is found, the status is `????`. My proposal is `TIME LIMIT REACHED` or `INTEGER FEASIBLE`.

**Comment by jajhall** [comment]
If the MIP gap is reached, then we follow Gurobi in stating that the model status is `OPTIMAL`. After all, as Gurobi put in their definition of `Optimal` in the [Optimization Status Codes](https://www.gurobi.com/documentation/10.0/refman/optimization_status_codes.html) 
"Model was solved to optimality (subject to tolerances), and an optimal solution is available." For us and, presumably, Gurobi, the MIP gap is another tolerance on optimality.

I agree that, when solved with `time_limit` the status should be `TIME LIMIT REACHED`. I'll fix that. I'll also see what is returned as `solution` if an integer feasible point has been found before the time limit is reached.

**Comment by MikeSteglich** [comment]
Thank you. It helps me to implement and test the HiGHS interface to [CMPL](coliop.org).

**Comment by jajhall** [comment]
> If it is solved with time_limit and a solution is found, the status is `????`.

Which status is `????`: the `info.primal_solution_status`?

I've also found that when solved with time_limit the `model_status` seems to be `TIME LIMIT REACHED`, and if an integer feasible point has been found before the time limit is reached, then it is available to the user.

**Comment by MikeSteglich** [comment]
After running the HiGHS binary from CMPL, the solution file (`write_solution_style = 2`) is parsed. 

For example ...
```
c Problem:    _cmpl_28602
c Rows:       1563
c Columns:    1639
c Non-zeros:  9246
c Status:     ????
c Objective:  line_1 = 523.743594 (MINimum)
c
s mip 1563 1639 ? 523.743594
i 1 523.743594
i 2 1
...
```

**Comment by jajhall** [comment]
I returned `????` (should be `???`) in the Glpsol solution file because that's what I thought GLPK would do in cases like time-out. I think I see now what GLPK would do if time-our occurred.

**Comment by MikeSteglich** [comment]
I found for mip solutions the following in the Glpk manual:

```
Solution line. There must be exactly one solution line in the solution file. This line must appear before any other lines except comment lines and has the following format:
   s mip ROWS COLS SST OBJ
The lower-case letter s specifies that this is the solution line.
The three-character solution designator mip identifies the file as containing a solution to the MIP problem instance.
The ROWS and COLS fields contain non-negative integer values that specify the number of rows (constraints) and columns (variables), resp., in the LP problem instance.
The SST field contains one of the following lower-case letters that specifies the solution status: o — solution is integer optimal;
f — solution is integer feasible (non-optimal);
n — no integer feasible solution exists;
u — solution is undefined.
The OBJ field contains a floating-point number that specifies the objective function value in the
MIP solution.

```
But HiGHS uses also comment lines for status information. Could not the line `c Status: ` used for more detailed information.

**Comment by jajhall** [comment]
Fix is now merged into `latest`, which will become `v1.5.0` in due course
